### ISSUE TITLE
could call hierarchy view show function signature, such as return or parameter type?

### ISSUE THREAD

**Issue by tarik-jin** [clangd/clangd#915]
![image](https://user-images.githubusercontent.com/39551684/139696213-cf3e84e6-60e5-4512-99af-96035adce7be.png)
just wonder if something like fun2(int) can be shown in call hierarchy

**Comment by HighCommander4** [comment]
Yes, I think that's a good idea. `CallHierarchyItem` in the [LSP spec](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_prepareCallHierarchy) has a `detail` field which could be used for this purpose (its description even says "e.g. the signature of a function"). Clangd does not currently populate this field.

**Comment by tarik-jin** [comment]
@HighCommander4 another suggestion:
it seems when click a function in call hierarchy view, editor will jump to first line in function definition, instead of call site, for example, when I click main function in call hierarchy, focus will jump to line 30 where main function definition, what about a setting option that let user decide whether jumps to definition function(line 30) or call site(line 40), seems when caller function is large,  jump directly to call site more reasonable(highlight call site depends on color theme, sometimes seems not visible)
![image](https://user-images.githubusercontent.com/39551684/139773780-e33dfd99-2c9e-48e0-8f4d-23951661b662.png)

**Comment by HighCommander4** [comment]
@tarik-jin the second issue you mention is on the vscode side, it's not within clangd's control. I have filed https://github.com/microsoft/vscode/issues/111454 about it a while ago.

**Comment by tarik-jin** [comment]
> @tarik-jin the second issue you mention is on the vscode side, it's not within clangd's control. I have filed [microsoft/vscode#111454](https://github.com/microsoft/vscode/issues/111454) about it a while ago.

got it, thanks a lot

**Comment by HighCommander4** [comment]
I started implementing this, but realized that the index does not store function signatures for "main-file only" functions (functions defined in .cpp files with no declaration in a header file), and such functions do tend to show up in call hierarchy trees.

Should we start storing function signatures for such functions?

**Comment by kadircet** [comment]
That actually sounds like a worthy tradeoff nowadays as more and more editors are implementing the feature. You (or whomever is interested in chasing this further) mind changing the logic in SymbolCollector to calculate && set signatures of symbols before bailing out for non-code completion symbols and posting some stats around the increase in both disk-size && mem-usage by background-index for LLVM codebase?
